From 3f5fa933d17e24cafd75ae28c00f4fc0292eae92 Mon Sep 17 00:00:00 2001
From: Johannes Huber <johu@gentoo.org>
Date: Wed, 15 Aug 2012 14:42:16 +0200
Subject: [PATCH] Set some proper paths to make cmake find our tools.

The ebuild now adds an extra / at the end of $EPREFIX so that it is
never the empty string (so that CMAKE_SYSTEM_PREFIX_PATH remains
correct)

Original patch by Heiko Przybyl.
Updated by Chris Reffett (cmake-2.8.8)
Update by Johannes Huber (cmake-2.8.9)
---
 Modules/Platform/Darwin.cmake    |   12 +++++++----
 Modules/Platform/UnixPaths.cmake |   43 +++++++++++++++++++-------------------
 2 files changed, 29 insertions(+), 26 deletions(-)

diff --git a/Modules/Platform/Darwin.cmake b/Modules/Platform/Darwin.cmake
index eff07b2..cbedc47 100644
--- a/Modules/Platform/Darwin.cmake
+++ b/Modules/Platform/Darwin.cmake
@@ -228,24 +228,28 @@ SET(CMAKE_CXX_CREATE_MACOSX_FRAMEWORK
       "<CMAKE_CXX_COMPILER> <LANGUAGE_COMPILE_FLAGS> <CMAKE_SHARED_LIBRARY_CREATE_CXX_FLAGS> <LINK_FLAGS> -o <TARGET> <SONAME_FLAG> <TARGET_INSTALLNAME_DIR><TARGET_SONAME> <OBJECTS> <LINK_LIBRARIES>")
 
 
-# default to searching for frameworks first
+# default to searching for frameworks last
 IF(NOT DEFINED CMAKE_FIND_FRAMEWORK)
-  SET(CMAKE_FIND_FRAMEWORK FIRST)
+  SET(CMAKE_FIND_FRAMEWORK LAST)
 ENDIF()
 # set up the default search directories for frameworks
 SET(CMAKE_SYSTEM_FRAMEWORK_PATH
+  @GENTOO_PORTAGE_EPREFIX@Frameworks
+  @GENTOO_PORTAGE_EPREFIX@usr/lib
   ~/Library/Frameworks
   /Library/Frameworks
   /Network/Library/Frameworks
   /System/Library/Frameworks)
 
-# default to searching for application bundles first
+# default to searching for application bundles last
 IF(NOT DEFINED CMAKE_FIND_APPBUNDLE)
-  SET(CMAKE_FIND_APPBUNDLE FIRST)
+  SET(CMAKE_FIND_APPBUNDLE LAST)
 ENDIF()
 # set up the default search directories for application bundles
 SET(_apps_paths)
 FOREACH(_path
+  @GENTOO_PORTAGE_EPREFIX@Applications
+  @GENTOO_PORTAGE_EPREFIX@usr/bin
   "~/Applications"
   "/Applications"
   "${OSX_DEVELOPER_ROOT}/../Applications" # Xcode 4.3+
diff --git a/Modules/Platform/UnixPaths.cmake b/Modules/Platform/UnixPaths.cmake
index 5ee7ddb..961746d 100644
--- a/Modules/Platform/UnixPaths.cmake
+++ b/Modules/Platform/UnixPaths.cmake
@@ -33,6 +33,7 @@ GET_FILENAME_COMPONENT(_CMAKE_INSTALL_DIR "${_CMAKE_INSTALL_DIR}" PATH)
 # search types.
 LIST(APPEND CMAKE_SYSTEM_PREFIX_PATH
   # Standard
+  @GENTOO_PORTAGE_EPREFIX@usr/local @GENTOO_PORTAGE_EPREFIX@usr @GENTOO_PORTAGE_EPREFIX@
   /usr/local /usr /
 
   # CMake install location
@@ -44,43 +45,41 @@ LIST(APPEND CMAKE_SYSTEM_PREFIX_PATH
 
 # List common include file locations not under the common prefixes.
 LIST(APPEND CMAKE_SYSTEM_INCLUDE_PATH
-  # Windows API on Cygwin
-  /usr/include/w32api
-
-  # X11
-  /usr/X11R6/include /usr/include/X11
-
-  # Other
-  /usr/pkg/include
-  /opt/csw/include /opt/include  
-  /usr/openwin/include
+  @GENTOO_PORTAGE_EPREFIX@usr/include
   )
 
 LIST(APPEND CMAKE_SYSTEM_LIBRARY_PATH
-  # Windows API on Cygwin
-  /usr/lib/w32api
-
-  # X11
-  /usr/X11R6/lib /usr/lib/X11
-
-  # Other
-  /usr/pkg/lib
-  /opt/csw/lib /opt/lib 
-  /usr/openwin/lib
+  @GENTOO_PORTAGE_GCCLIBDIR@/gcc
+  @GENTOO_PORTAGE_GCCLIBDIR@
+  @GENTOO_PORTAGE_EPREFIX@usr/lib64
+  @GENTOO_PORTAGE_EPREFIX@usr/libx32
+  @GENTOO_PORTAGE_EPREFIX@usr/lib32
+  @GENTOO_PORTAGE_EPREFIX@usr/lib
+  @GENTOO_PORTAGE_EPREFIX@lib
   )
 
 LIST(APPEND CMAKE_SYSTEM_PROGRAM_PATH
-  /usr/pkg/bin
+  @GENTOO_PORTAGE_EPREFIX@usr/bin
+  @GENTOO_PORTAGE_EPREFIX@bin
   )
 
 LIST(APPEND CMAKE_PLATFORM_IMPLICIT_LINK_DIRECTORIES
-  /lib /usr/lib /usr/lib32 /usr/lib64
+  @GENTOO_PORTAGE_GCCLIBDIR@/gcc
+  @GENTOO_PORTAGE_GCCLIBDIR@
+  @GENTOO_PORTAGE_EPREFIX@usr/lib64
+  @GENTOO_PORTAGE_EPREFIX@usr/libx32
+  @GENTOO_PORTAGE_EPREFIX@usr/lib32
+  @GENTOO_PORTAGE_EPREFIX@usr/lib
+  @GENTOO_PORTAGE_EPREFIX@lib
+  /lib /usr/lib /usr/lib32 /usr/lib64 /usr/libx32
   )
 
 LIST(APPEND CMAKE_C_IMPLICIT_INCLUDE_DIRECTORIES
+  @GENTOO_PORTAGE_EPREFIX@usr/include
   /usr/include
   )
 LIST(APPEND CMAKE_CXX_IMPLICIT_INCLUDE_DIRECTORIES
+  @GENTOO_PORTAGE_EPREFIX@usr/include
   /usr/include
   )
 
-- 
1.7.9.7

